users/vieting/experiments/switchboard/ctc/feat/network_helpers/perturbation.py

```diff
@@ -58,60 +54,51 @@ def __init__(
         """
         self._speed = PerturbationFactor(**speed) if speed else None
         self._tempo = PerturbationFactor(**tempo) if tempo else None
-        self._perturbations = [functools.partial(self.sox, sox_effects=sox_effects)]
+        self._sox_effects = sox_effects or []
         if preemphasis:
             self._perturbations.append(functools.partial(self.preemphasis, factor=PerturbationFactor(**preemphasis)))
-        if codecs:
-            self._perturbations.append(functools.partial(self.apply_codecs, codecs=codecs))
 
     def run(self, audio, sample_rate, random_state):
-        audio = torch.from_numpy(audio).unsqueeze(0).to(torch.float32)
-        for perturbation in self._perturbations:
-            audio = perturbation(audio, sample_rate, random_state)
-        audio = audio.numpy().squeeze()
+        import numpy as np
```

Comment:
The numpy import should already be at the top of the file given the code in `get_code_for_perturbation()`, right?

Reply:
Done